src/runtime/pipeline/pipeline_function.cc

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "pipeline_function.h"
+
+#include <utility>
+#include <vector>
+namespace tvm {
+namespace runtime {
+/*!
+ * \brief Initialize pipeline.
+ * \param modules  List of graph runtime module.
+ * \param pipeline_conf Dependency relation of each graph runtime module.
+ * \param mod_configure Configure information that generate by export library function call.
+ */
+size_t PipelineFunction::PipelineInit(Array<Module> modules, const PipelineConfigure& pipeline_conf,
+                                      const ModuleConfigure& mod_configure) {
+  int outputNum = pipeline_conf.GetGlobalOutputNum();
+  std::vector<Module> graphRuntimes = PipelineCreateGraphruntime(modules, mod_configure);
+  return outputNum;
+}
+/*!
+ * \brief There are two mode to create graph runtime list, first is to use modules that
+ *  are the module list already created by caller, when modules is empty these information
+ *  from mod_configure will get use to create graph runtime list.
```

Comment:
I think it is better to separate the two cases in different function, since the first one is much simpler. Actually we don't need a dedicated function for the first case.

Reply:
fixed.